.github/workflows/macos.yml

```diff
@@ -52,6 +63,8 @@ jobs:
         with:
           sparse-checkout-cone-mode: false
           sparse-checkout: /.github
+          # Set fetch-depth: 0 so that Launchable can receive commits information.
+          fetch-depth: 0
```

Comment:
Checking the [actions/checkout](https://github.com/actions/checkout) repository's README, the default value is 1. Now the value is 0.

And I compared the macOS - macos-12 case log between the [current master branch's log](https://github.com/ruby/ruby/actions/runs/7905688154/job/21578858298#step:2:56) (the actions/checkout process taking 3 seconds) and [this PR's log](https://github.com/ruby/ruby/actions/runs/7885458586/job/21516643427?pr=9777#step:2:60) (the actions/checkout process taking 3 seconds).

The `git fetch` log in the current master branch with the default `fetch-depth: 1` is below. You see the `--no-tags` option is used.

```
/usr/local/bin/git -c protocol.version=2 fetch --no-tags --prune --no-recurse-submodules --filter=blob:none --depth=1 origin +fc2c128e7e1fd4a197669d2f04e8457f4188b7f3:refs/remotes/origin/master
```

The `git fetch` log in this PR with `fetch-depth: 0` is below. The actions/checkout process takes 13 seconds. You see the `--no-tags` option disappeared in the command line. And fetching all the branches and tags.

```
/usr/local/bin/git -c protocol.version=2 fetch --prune --no-recurse-submodules --filter=blob:none origin +refs/heads/*:refs/remotes/origin/* +refs/tags/*:refs/tags/* +60cc647c1983035c7998d1a6fbfd1c832aea3e96:refs/remotes/pull/9777/merge
```

Does Launchable need to get the branch and tags information too? Can we add the `fetch-tags: false` explicitly maybe that is an option to add the `--no-tags` option to the `git fetch` to save the running time?

```
fetch-tags: false
```

Reply:
Updated

Reply:
Sorry, I found that adding the `fetch-tags: false` doesn't mean adding the `--no-tags` option seeing the PR's latest CI log, and the command.
https://github.com/ruby/ruby/actions/runs/7912934050/job/21599616384?pr=9777#step:2:60

```
/usr/local/bin/git -c protocol.version=2 fetch --prune --no-recurse-submodules --filter=blob:none origin +refs/heads/*:refs/remotes/origin/* +refs/tags/*:refs/tags/* +dc79308e09e64764d06e6f718a3d54174049e77a:refs/remotes/pull/9777/merge
```

And below is the previous log before you rebase this PR. The `fetch-tags: false` is already used.
https://github.com/ruby/ruby/actions/runs/7885458586/job/21516643427?pr=9777#step:2:11

I found the part of adding the `--no-tags` option in the actions/checkout's source. However, I don't understand the condition.

https://github.com/actions/checkout/blob/b4ffde65f46336ab88eb53be808477a3936bae11/src/git-command-manager.ts#L253-L255

```
    if (!refSpec.some(x => x === refHelper.tagsRefSpec) && !options.fetchTags) {
      args.push('--no-tags')
    }
```

So, please remove the line `fetch-tags: false`. Sorry for that.

Reply:
No problem. Updated

Reply:
Thanks. I confirmed that the `fetch-tags: false` was removed on the latest commit in this PR.